### PR TITLE
Add dependency to opensearch-ruby

### LINKED ISSUE #55: Test against logstash 7.16.1
Update workflow to include 7.16.1

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
* Use OpenSearch::Client instead of Elasticsearch.
* Update CI to test last two minor release.

### Issues Resolved
#55 

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has documentation added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/integration/outputs/delete_spec.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@
       expect(r['_source']['message']).to eq('foo')
 
       subject.multi_receive([LogStash::Event.new("my_id" => id, "my_action" => "delete", "message" => "foo", "my_version" => 100)])
-      expect { es.get(:index => 'logstash-delete', :type => doc_type, :id => id, :refresh => true) }.to raise_error(Elasticsearch::Transport::Transport::Errors::NotFound)
+      expect { es.get(:index => 'logstash-delete', :type => doc_type, :id => id, :refresh => true) }.to raise_error(OpenSearch::Transport::Transport::Errors::NotFound)
```
Comment (jmazanec15) [inline spec/integration/outputs/delete_spec.rb:69]: Why is it still es?

File: spec/integration/outputs/templates_spec.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
   before :each do
     # Delete all templates first.
-    require "elasticsearch"
+    require "opensearch"
```
Comment (jmazanec15) [inline spec/integration/outputs/templates_spec.rb:25]: Why does this need to be required at all if no code references it?

File: spec/integration/outputs/delete_spec.rb (hunk 1 above)
Comment (VijayanB) [reply spec/integration/outputs/delete_spec.rb]: Found some more instance where variable name is not rename to opensearch. I updated it too.

File: spec/integration/outputs/templates_spec.rb (hunk 2 above)
Comment (VijayanB) [reply spec/integration/outputs/templates_spec.rb]: Ack.

**Review by naveentatikonda — APPROVED** [review]

**Review by jmazanec15 — APPROVED** [review]

**Pull request merged**